### PR TITLE
Xeno abilities easier to land on targets

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Add a lot of target flags on xeno abilities, so it's easier to hit the desired target. Hitting the turf where the target is will automaticly redirect your ability on the target

Remove unused targeting flags

## Why It's Good For The Game

With lag, byond being bad, high TD, you sometimes have to spam your abilities to manage to hit you target. This makes it easier to hit those, which is nice QOL

## Changelog
:cl:
balance: Hitting the turf where the target is will automaticly redirect your ability on the target. This works for most xeno ability
code: Remove unused targeting flags
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	ability_name = "plow charge"
 	keybind_signal = COMSIG_XENOABILITY_BULLCHARGE
 	var/new_charge_type = CHARGE_BULL
+	target_flags = XABB_HUMAN_TARGET
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm:11]: ?????????

File: code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm [hunk 2]
```diff
@@ -29,6 +30,7 @@
 	ability_name = "headbutt charge"
 	keybind_signal = COMSIG_XENOABILITY_BULLHEADBUTT
 	new_charge_type = CHARGE_BULL_HEADBUTT
+	target_flags = XABB_MOB_TARGET
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm:33]: ??????????????

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
these two skills are not targetted

**Comment by BraveMole** [comment]
> these two skills are not targetted

they are not? I didn't play bull in a very long time

File: code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm (hunk 2 above)
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm:33]:
```suggestion
```

File: code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm (hunk 1 above)
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm:11]:
```suggestion
```

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -315,6 +315,8 @@ if(selected_ability.target_flags & flagname){\
 		return;}}
 
 /mob/living/carbon/xenomorph/proc/ability_target(atom/A)
+	TARGET_FLAGS_MACRO(XABB_HUMAN_TARGET, /mob/living/carbon/human)
+	TARGET_FLAGS_MACRO(XABB_XENO_TARGET, /mob/living/carbon/xenomorph)
```
Comment (TiviPlus) [inline code/_onclick/click.dm:319]: Why this can just use XABB_MOB_TARGET

Comment (BraveMole) [reply code/_onclick/click.dm:319]: It's not impossible than an incoucious human or xeno are on the same turf than a consiouc mob that you really want to target.

This probably happens rarely, but it's possible. Why not preventing that issue at least partly? Some abilities cannot target a human, let's try to find only xenos then

Comment (TiviPlus) [reply code/_onclick/click.dm:319]: Because it adds extra processing on every single ability use for an edge case that will basically never happen

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**